libqtile/widget/window_count.py

```diff
@@ -55,19 +56,28 @@ def _setup_hooks(self):
 
     def _wincount(self, *args):
         try:
-            self._count = len(self.qtile.current_group.windows)
+            if this_screen:
```

Comment:
I think most of the if statements can go and this should just be updated to show the count on the relevant screen,